Source/Manager.swift

```diff
@@ -9,144 +9,159 @@
-        for (index, component) in pathComponents.enumerate() {
-
-            // if it does...
-            if let id = Int(component) {
-
-                // add it to the IDs array to be returned
-                if ids?.append(id) == nil {
-                    ids = [id]
-                }
-                
-                // remove the number and replace with predetermined key to be used for the name to be returned
-                pathComponents[index] = ":id"
-            }
-        }
 
-        // prevents "//" from occurring when joining the path components
-        if pathComponents.first == separator {
-            pathComponents.removeFirst()
-        }
-        
-        name = pathComponents.joinWithSeparator(separator)
-        name = separator + name!
+    name = pathComponents.joinWithSeparator(separator)
+    name = separator + name!
 
-        let query = queryDictionaryFromURL(url)
-        
-        return (name!, ids, query)
+    let body = ArtifactDictionaryFromData(request.HTTPBody)
+    let query = ArtifactDictionaryFromString(url.query)
+
+    return (name!, ids, query, body)
+  }
+
+  /// Return a dictionary with arrays populated with the values of HTTP body data. URLs allow keys to be declared multiple times, hence the array container.
+  static func ArtifactDictionaryFromData(data: NSData?) -> QueryArtifacts? {
+    guard let data = data,
+      let componentString = String(data: data, encoding: NSUTF8StringEncoding)
+      else {
+        return nil
     }
 
-    // inspired by https://gist.github.com/freerunnering/1215df277d750af71887
-    /// Return a dictionary with arrays populated with the values of query parameters. URLs allow keys to be declared multiple times, hence the array container.
-    static func queryDictionaryFromURL(url: NSURL) -> QueryArtifacts? {
-        guard let query = url.query else { return nil }
+    return ArtifactDictionaryFromString(componentString)
+  }
 
-        var queryDict = QueryArtifacts()
+  // inspired by https://gist.github.com/freerunnering/1215df277d750af71887
+  /// Return a dictionary with arrays populated with the values of query parameters. URLs allow keys to be declared multiple times, hence the array container.
+  static func ArtifactDictionaryFromString(string: String?) -> QueryArtifacts? {
+    guard let string = string else { return nil }
 
-        for kVString in query.componentsSeparatedByString("&") {
-            let parts = kVString.componentsSeparatedByString("=")
+    var queryDict = QueryArtifacts()
 
-            guard parts.count > 1 else { continue }
+    for kVString in string.componentsSeparatedByString("&") {
+      let parts = kVString.componentsSeparatedByString("=")
 
-            let key = parts.first!.stringByRemovingPercentEncoding!
-            let value = parts.last!.stringByRemovingPercentEncoding!
-            var values = queryDict[key] ?? [String]()
+      guard parts.count > 1 else { continue }
```

Comment:
I don't know if this use of guard is brilliant or insane
for me `if parts.count > 1 { ... }` seems easier to visually parse, but that may be because guard is new to me